sdk/core/core/inc/az_pair.h

```diff
@@ -13,23 +13,38 @@
 
 #include <_az_cfg_prefix.h>
 
-/// A pair of strings.
+/**
+ * @brief a pair of az_span of bytes as a key and value
+ *
+ */
 typedef struct {
   az_span key;
   az_span value;
 } az_pair;
 
-/// A span of pairs.
+/**
+ * @brief an immutable span of az_pairs where begin points to the first az_pair
+ * and size represents the total number of az_pairs
+ */
 typedef struct {
   az_pair const * begin;
   size_t size;
 } az_pair_span;
 
+/**
+ * @brief a mutable span of az_pairs where begin points to the first az_pair
+ * and size represents the total number of az_pairs
+ */
 typedef struct {
   az_pair * begin;
   size_t size;
 } az_mut_pair_span;
 
+/**
+ * @brief a builder for construnction an az_pair_span.
+ * @var buffer.size represents the maximun size of az_pair_span that can be added
+ * @var size represents the current number of az_pair_span in the buffer
```

Comment:
I still don't feel great about the relationship between a span and a span_builder. Maybe, we should have span_builder struct which has a begin, size, and length (used size). Then an _init function that takes these 3 parameters and a function that takes a span_builder and return the used_span (as a span).

Having a span_builder that contains a span and then having 2 size fields is not a good conceptual model.

Reply:
https://github.com/Azure/azure-sdk-for-c/pull/284 addressed updating all builders to have a `length` field instead of `size`.
IMHO, having builder with begin and size looks redundant if this is exactly what a az_span is.  Think in a weird case when az_span would change, that would be a major change to update all builders.